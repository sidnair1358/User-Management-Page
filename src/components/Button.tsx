import { ButtonProps } from "../types";

const Button = ({
  children,
  label,
  onClick,
  type,
  disabled = false,
}: ButtonProps) => {
  let styles =
    "rounded bg-green-500 px-4 py-2 text-white disabled:opacity-50 hover:bg-green-400";
  if (label === "submit") {
    styles = "rounded bg-blue-500 px-4 py-2 text-white hover:bg-blue-400";
  } else if (label === "dismiss") {
    styles = "absolute right-2 top-2";
  }

  return (
    <button
      onClick={onClick}
      type={type}
      aria-label={label}
      disabled={disabled}
      className={styles}
    >
      {children}
    </button>
  );
};

export default Button;
