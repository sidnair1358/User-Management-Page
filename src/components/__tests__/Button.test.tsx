import { render } from "@testing-library/react";
import Button from "../Button";

describe("Button Component", () => {
  test("Renders correctly for default button", () => {
    const { asFragment } = render(
      <Button label="default" type="button" onClick={() => {}}>
        Button Text
      </Button>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
