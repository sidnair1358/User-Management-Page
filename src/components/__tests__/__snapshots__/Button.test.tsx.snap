// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Renders correctly for default button 1`] = `
<DocumentFragment>
  <button
    aria-label="default"
    class="rounded bg-green-500 px-4 py-2 text-white disabled:opacity-50 hover:bg-green-400"
    type="button"
  >
    Button Text
  </button>
</DocumentFragment>
`;

exports[`Button Component renders correctly for default button 1`] = `
<DocumentFragment>
  <button
    aria-label="default"
    class="rounded bg-green-500 px-4 py-2 text-white disabled:opacity-50 hover:bg-green-400"
    type="button"
  >
    Button Text
  </button>
</DocumentFragment>
`;
